### PR TITLE
Shell: Allow redirections and pipes on builtins

### LINKED ISSUE #3072: `time` doesn't play well with pipes
`disasm /bin/id | head` prints 10 lines of output, as it should.

`time disasm /bin/id | head` doesn't limit output. It acts as if the `| head` bit didn't exist.

@alimpfard

### PR DESCRIPTION
**Pull request by alimpfard**
Fixes #3072.

While this does indeed fix that, `time` will still fail to play well with pipes, as the command will run to completion before any other piped command after it is even invoked.
e.g.
```sh
time long-running-command foo bar | head
```
will wait for the entirety of `long-running-command`'s runtime before invoking `head`.

### CODE REVIEW

File: Shell/main.cpp [hunk 1]
```diff
@@ -56,6 +56,31 @@ void FileDescriptionCollector::add(int fd)
     m_fds.append(fd);
 }
 
+SavedFileDescriptors::SavedFileDescriptors(const NonnullRefPtrVector<AST::Rewiring>& intended_rewirings)
+{
+    for (auto& rewiring : intended_rewirings) {
+        int new_fd = fcntl(0, F_DUPFD, rewiring.source_fd);
```
Comment (alimpfard) [inline Shell/main.cpp:62]: This is one weird interface...
`fcntl` already takes an fd as its first arg, _why_ would anyone think "hey, let's just ignore the perfectly fine fd argument and ask for a third argument for `F_DUPFD`"? hm Dr. POSIX? hm?

Comment (alimpfard) [reply Shell/main.cpp:62]: Hmmm...this needs `CLOEXEC`

Comment (awesomekling) [reply Shell/main.cpp:62]: Uh what's wrong with using `dup()` here?

Comment (alimpfard) [reply Shell/main.cpp:62]: Nothing wrong with `dup()`, I just wanted to whine about how weird `fcntl` is lol

**Pull request merged**